falcon/testing/test_case.py

```diff
@@ -95,10 +95,11 @@ def text(self):
                 self._text = u''
             else:
                 if self.encoding is None:
-                    msg = 'Response did not specify a content encoding'
-                    raise RuntimeError(msg)
+                    encoding = 'UTF-8'
+                else:
+                    encoding = self.encoding
 
-                self._text = self.content.decode(self.encoding)
+                self._text = self.content.decode(encoding)
```

Comment:
you can replace your changes with:
self._text = self.content.decode(self.encoding or 'UTF-8')